### ISSUE TITLE
8266 WebSockets Server

### ISSUE THREAD

**Issue by FabRo57** [Links2004/arduinoWebSockets#657]
Hello , when accessing from a portable device , it often happens I close my app whithout disconnecting first from the ESP8266 so  every new access from the smartphone increases the counter and when the max number of client is reached , the system becomes unresponsive from remote : does the library recognise a command like 'disconnect' or can I set a timeout and , from inside the program running on the ESP , close in some way ( or reset to 0 ) all the active connections without a reboot ?

Hope I explained in not too messy way, thanks for help .

FabRo

**Comment by Links2004** [comment]
yes, you can activate a timeout (keep alive / heartbeat) or manually disconnect all or one client.

https://github.com/Links2004/arduinoWebSockets/blob/a14b6b73b4f05e189ca49d6e84202c2b55db528a/src/WebSocketsServer.h#L90-L91

https://github.com/Links2004/arduinoWebSockets/blob/a14b6b73b4f05e189ca49d6e84202c2b55db528a/src/WebSocketsServer.h#L80-L81

**Comment by FabRo57** [comment]
Thank you so much. Inviato dal mio Galaxy
-------- Messaggio originale --------Da: Markus ***@***.***> Data: 18/04/21  19:54  (GMT+01:00) A: Links2004/arduinoWebSockets ***@***.***> Cc: FabRo57 ***@***.***>, Author ***@***.***> Oggetto: Re: [Links2004/arduinoWebSockets] 8266 WebSockets Server (#657) 
yes, you can activate a timeout (keep alive / heartbeat) or manually disconnect all or one client.
https://github.com/Links2004/arduinoWebSockets/blob/a14b6b73b4f05e189ca49d6e84202c2b55db528a/src/WebSocketsServer.h#L90-L91
https://github.com/Links2004/arduinoWebSockets/blob/a14b6b73b4f05e189ca49d6e84202c2b55db528a/src/WebSocketsServer.h#L80-L81

**Comment by FabRo57** [comment]
Hi , I thank you again , but I have no clear ideas on how to use your suggestions : I guess (!)  the  enableHeartbeat() function should test if a certain client# is still connected , otherwise it should automatically disconnect it ? 
It is my unexperience , but I can's see how I could link that function to a certain client# , and how can I test if it is really checking client presence.
Any further info would be greatly appreciated. thank you .
FR
